extension/maxminddb.c

```diff
@@ -107,9 +107,64 @@ static int Reader_init(PyObject *self, PyObject *args, PyObject *kwds)
 static PyObject *Reader_get(PyObject *self, PyObject *args)
 {
     char *ip_address = NULL;
+    PyObject *object = NULL;
+#if PY_MAJOR_VERSION >= 3
+    PyObject *bytes = NULL;
+#endif
 
     Reader_obj *mmdb_obj = (Reader_obj *)self;
-    if (!PyArg_ParseTuple(args, "s", &ip_address)) {
+
+    if (PyArg_ParseTuple(args, "s", &ip_address)) {
+        // pass
+    } else if (PyErr_Clear(), PyArg_ParseTuple(args, "O", &object)) {
+        PyObject* module = PyImport_ImportModule("ipaddress");
+        if (module == NULL) {
+            return NULL;
+        }
+
+        PyObject* module_dict = PyModule_GetDict(module);
+        if (module_dict == NULL) {
+            return NULL;
+        }
+
+        PyObject* ipaddress_IPv4Address = PyDict_GetItemString(
+            module_dict, "IPv4Address");
+        int is_ipaddress_object = 0;
+
+        if (ipaddress_IPv4Address != NULL) {
+            is_ipaddress_object = PyObject_IsInstance(ipaddress_IPv4Address,
+                object);
+        }
+
+        PyObject* ipaddress_IPv6Address;
+        if (!is_ipaddress_object &&
+            (ipaddress_IPv6Address = PyDict_GetItemString(module_dict,
+                "IPv6Address")) != NULL) {
+            is_ipaddress_object = PyObject_IsInstance(ipaddress_IPv6Address,
+                object);
+        }
+
+        PyErr_Clear();
+        PyObject *str;
+
+        if (!is_ipaddress_object) {
+            PyErr_SetString(PyExc_TypeError, "IP address must be a string,"
+                " ipaddress.IPv4Address or ipaddress.IPv6Address object");
+        } else if ((str = PyObject_Str(object)) != NULL) {
```

Comment:
It is a little bit weird that we are doing all of these gymnastics and creating these heavy Python string and byte objects. I wonder if we could just get the packed address. Ideally, we wouldn't even bother converting that back to a *char, but it probably doesn't hurt too much to do that conversion.

Reply:
What's your non-weird proposal? The only other method I could think of would be casting the argument to a `str`, but then you lose out on type safety.

Reply:
> I wonder if we could just get the packed address

That is exposed:

```
~$ python
Python 3.5.3 (default, Sep 27 2018, 17:25:39)
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ipaddress
>>> i = ipaddress.ip_address("127.0.0.1")
>>> i.packed
b'\x7f\x00\x00\x01'
```